src/healthcheck/HealthChecker.ts

```diff
@@ -118,51 +122,58 @@ class HealthChecker {
 
   public async getStatus(): Promise<HealthStatus> {
     if (this.shutdownRequested === true) {
-      return this.getShutdownStatus()
+      return this.getShutdownStatus();
     }
     if (this.startupComplete === false) {
-      return this.getStartupStatus()
+      return this.getStartupStatus();
     }
-    return this.getHealthStatus()
+    return this.getHealthStatus();
+  }
+
+  private async getPromiseStatus(statusResponse: HealthStatus) {
+    const runChecks = this.startupPlugins.map(check => check.runCheck());
+    Promise.all(runChecks);
```

Comment:
`await` is missing here. Without `await`, the next statement will be executed by the `runCheck` is done. As a result, the statusResponse contains incorrect statuses.

Reply:
Originally left the await out as it failed quite a few tests. Have however re-looked at the tests and made some changes. I have now added the await.